boot/bootutil/src/loader.c

```diff
@@ -49,7 +49,9 @@
 #include "bootutil/mcuboot_status.h"
 
 #if defined(CONFIG_SOC_NRF5340_CPUAPP) && defined(PM_CPUNET_B0N_ADDRESS)
+#include <fw_info_bare.h>
 #include <dfu/pcd.h>
+int pcd_version_cmp_net(const struct flash_area *fap, struct image_header *hdr);
```

Comment:
Question: wouldn't it be more useful if the function would rather extract version out of PCD and comparison would happen on the MCUboot side?

Reply:
Doesn't use the same versioning scheme. Netcore uses FW_INFO which is just a single value while MCUBoot does the major.minor.rev semantic versioning.

The end goal is to consolidate them but for now, this is what we can provide. 

The command is running on the MCUBoot side but it fetches the current version number through PCD from the network core and does the comparison on Application(MCUBoot side).